libs/klv/src/main/java/org/codice/ddf/libs/klv/data/text/KlvEncodingDetectedString.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.libs.klv.data.text;
+
+import java.io.UnsupportedEncodingException;
+import org.apache.tika.parser.txt.CharsetDetector;
+import org.apache.tika.parser.txt.CharsetMatch;
+import org.codice.ddf.libs.klv.KlvDataElement;
+import org.codice.ddf.libs.klv.data.Klv;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Represents a KLV data element that has a <strong>String</strong> value. The string may not be the
+ * machine's encoding, so it should be detected
+ */
+public class KlvEncodingDetectedString extends KlvDataElement<String> {
+  private static final Logger LOGGER = LoggerFactory.getLogger(KlvEncodingDetectedString.class);
+
+  /**
+   * Constructs a {@code KlvDataElement} that describes how to interpret the value of a data element
+   * with the given key.
+   *
+   * @param key the data element's key
+   * @param name a name describing the data element's value
+   * @throws IllegalArgumentException if any arguments are null
+   */
+  public KlvEncodingDetectedString(byte[] key, String name) {
+    super(key, name);
+  }
+
+  @Override
+  protected void decodeValue(Klv klv) {
+    byte[] bytes = klv.getValue();
```

Comment:
❓  Are we guaranteed that the `klv` passed in will be not null?

Reply:
I can add a null check here.

Reply:
❓  What should we do if it's null? It doesn't look like we null checked anything else in this code so I'm not sure what would be best. 
I was thinking: null or an empty string.